common/hal/hal_i3c.c

```diff
@@ -371,11 +371,11 @@ int i3c_controller_write(I3C_MSG *msg)
 		LOG_ERR("Failed to write messages to bus 0x%d addr 0x%x, ret: %d", msg->bus,
 			msg->target_addr, ret);
 		k_mutex_unlock(&mutex_dev[msg->bus]);
-		return -1;
+		return false;
```

Comment:
Its generally not good practice to mix bool and ints. Having a function returning int return true/false is not good. Either we change the prototype or return ints.
We seem to have larger issues with the use of this API
Looking at the callers of i3c_controller_write, p3h284x.c, rg3mxxb12.c, 
they use 0 as error and non-zero as success. Almost like it is returning bool.

common/service/mctp/mctp_i3c.c is using negative values as error and 0 or positive as success. So, that might not get the error.
plat_pmic.c - is using 0 as success and non-zero as error.

hal_i3c.c seems to be doing a similar thing of mixing true/false with integer returns (bad practice).

Can you take a look at these methods and see if we can unify return codes. Either change the API to return bool or return int and define exactly what this function should return.